lib/galaxy/jobs/runners/pulsar.py

```diff
@@ -253,6 +253,8 @@ def check_watched_item(self, job_state):
 
     def _update_job_state_for_status(self, job_state, pulsar_status, full_status=None):
         if pulsar_status == "complete":
+            job_state.running = False
+            job_state.job_wrapper.change_state(model.Job.states.OK)
```

Comment:
I don't know why the Galaxy finish code isn't being called in your cases, but we really can't set OK here. There is a lot of code in jobs/runners/__ini__.py that still needs to run for these jobs - Galaxy needs to determine if they ran okay, needs to collect metadata, etc..

Maybe put some debugging in ``run_next`` in job runners to see if the jobs really do get queued for completion in the following line?